platforms/allay/build.gradle.kts

```diff
@@ -0,0 +1,16 @@
+repositories {
+    maven("https://repo.opencollab.dev/maven-releases/")
+    maven("https://repo.opencollab.dev/maven-snapshots/")
+    maven("https://storehouse.okaeri.eu/repository/maven-public/")
+    maven("https://jitpack.io/")
+}
+
+dependencies {
+    shadedApi(project(":common:implementation:base"))
+    implementation("com.google.code.gson", "gson", "2.11.0")
+
+    compileOnly("org.projectlombok:lombok:1.18.32")
+    compileOnly("org.allaymc.allay:api:master-SNAPSHOT")
+
+    annotationProcessor("org.projectlombok:lombok:1.18.32")
+}
```

Comment:
All of these versions should go in Versions.kt. Gson should go to Libraries internal as I plan to use it for other things and it's a good library to have set there. everything else should go in a a new allay section. Also is lombok necessary as part of the platform or is it just a convenience if it's not required I'd prefer if we moved away from it here as to not introduce any giant dependencies. Also we should pin a version of allay so a new version does not have to be downloaded every time you open the an ide to work on Terra.
Also follow the format (package, dep, version) like gson.